src/Columns/ColumnString.cpp

```diff
@@ -302,6 +302,43 @@ void ColumnString::getPermutation(bool reverse, size_t limit, int /*nan_directio
     }
 }
 
+void ColumnString::getPermutationHint(bool reverse, size_t limit, int /*nan_direction_hint*/, Permutation & res, const IColumn::Filter & hint) const
```

Comment:
I've resenctly added IColumn::getSpecialPermutation(). It looks like it's the same problem here. https://github.com/ClickHouse/ClickHouse/blob/master/src/Columns/IColumn.h#L261